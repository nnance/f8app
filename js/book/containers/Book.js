import gql from 'graphql-tag';
import { graphql } from 'react-apollo';
import _ from 'lodash';
import { createReducerOnReply, createReducerOnDelete } from 'graphql-comment/src/react-native';

import Book from '../components/Book';

export const query = gql`
  query BookQuery($id: MongoID!){
    clog(filter: { _id: $id }) {
      ...Book
    }
  }
  ${Book.fragments.clog}
`;

export const mapQueryToProps = ({ data }) => {
  const { clog, error, loading } = data;
  if (error) {
    console.error('graphql error: ', error);
  }
  return {
    clog: {
      ...(clog || {}),
      episodes: loading || error ? [] : clog.episodes,
    },
    loading,
  };
};

export const mapPropsToOptions = ({ id }) => {
  const replyReducer = createReducerOnReply(id, (previousResult, reply) => {
    return {
      ...previousResult,
      clog: {
        ...previousResult.clog,
        commentCount: previousResult.clog.commentCount + 1,
      },
    };
  });
  const deleteReducer = createReducerOnDelete(id, (previousResult) => {
    return {
      ...previousResult,
      clog: {
        ...previousResult.clog,
        commentCount: previousResult.clog.commentCount - 1,
      },
    };
  });
  return {
    variables: {
      id,
    },
    reducer: (previousResult, action) => {
      return replyReducer(deleteReducer(previousResult, action), action);
    },
  };
};

export default graphql(query, {
  props: mapQueryToProps,
  options: mapPropsToOptions,
})(Book);

// export default require('View')
