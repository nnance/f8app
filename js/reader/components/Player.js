import React from 'react';
import {
  Image,
  View,
  Text,
  TouchableOpacity,
  StyleSheet,
  StatusBar,
  Share,
  PanResponder,
  ActivityIndicator,
  Dimensions,
  Animated,
} from 'react-native';

import gql from 'graphql-tag';
import * as Progress from 'react-native-progress';

import { NavBarWithPinkButton } from '../../common/NavBar';
import { colors } from '../../common/styles';
import { toHumanNumber } from '../../common/utils';
import ModalSpinner from '../../common/ModalSpinner';
import BookAndPlayerTabBar from '../../common/BookAndPlayerTabBar';
import { serverURL } from '../../env';

// Import ClogPlayer component from submodule
// if any error occur please check if submodule is outdated
// note: please run `yarn install` inside player
// note2: react-native-fs require manual link `react-native link react-native-fs`
import ClogPlayer from '../../player/components/ClogPlayer';

const screenWidth = Dimensions.get('window').width;

const styles = StyleSheet.create({
  navButton: {
    width: 25,
    height: 25,
    marginRight: 10,
    resizeMode: 'contain',
  },
  titleText: {
    fontSize: 14,
    color: colors.textPink,
  },
  viewCountText: {
    fontSize: 10,
    color: colors.textFadedPink,
  },
  textContainer: {
    marginRight: 50,
  },
});

const MAX_HEIGHT_INDICATOR = 50;
const toMaxHeightIndicatorDuration = 200;
const hideIndicatorDuration = 300;

class Player extends React.Component {
  constructor(...args) {
    super(...args);
    this.state = {
      loading: true,
      dyPanResponder: 0,
      endedScrollView: false,
      nextEpisodeRatioAnimated: new Animated.Value(0.0),
      nextEpisodeRatioAnimatedValue: 0.0,
    };

    this.state.nextEpisodeRatioAnimated.addListener(({value}) => this.setState({
      nextEpisodeRatioAnimatedValue: value,
    }));

    this.onSharePress = this.onSharePress.bind(this);
    this.onCommentPress = this.onCommentPress.bind(this);
    this.onProgress = this.onProgress.bind(this);
    this.onBookmarkPress = this.onBookmarkPress.bind(this);
    this.onNextEpisode = this.onNextEpisode.bind(this);
    this.onRemoveBookmarkPress = this.onRemoveBookmarkPress.bind(this);
    this.renderNavBarButton = this.renderNavBarButton.bind(this);
    this.renderTitle = this.renderTitle.bind(this);

    this._panResponder = PanResponder.create({
      onStartShouldSetPanResponder: () => this.state.endedScrollView && this.props.episode.nextEpisode,
      onStartShouldSetPanResponderCapture: () => this.state.endedScrollView && this.props.episode.nextEpisode,
      onMoveShouldSetPanResponder: () => this.state.endedScrollView && this.props.episode.nextEpisode,
      onMoveShouldSetPanResponderCapture: () => this.state.endedScrollView && this.props.episode.nextEpisode,
      onPanResponderMove: (evt, gestureState) => {
        this.setState({
          dyPanResponder: gestureState.dy,
        });

        this.state.nextEpisodeRatioAnimated.setValue(this.nextEpisodeRatio());
      },
      onPanResponderRelease: () => {
        if (this.nextEpisodeRatio() >= 1) {
          this.onNextEpisode();
          Animated.timing(this.state.nextEpisodeRatioAnimated, {
            toValue: 1,
            duration: toMaxHeightIndicatorDuration,
          }).start();
        }
        else {
          Animated.timing(this.state.nextEpisodeRatioAnimated, {
            toValue: 0,
            duration: hideIndicatorDuration,
          }).start();
          this.setState({
            dyPanResponder: 0,
          });
        }
      },
    });
  }
  

  componentWillReceiveProps(nextProps) {
    if (this.props.loading === true && nextProps.loading === false) {
      this.onProgress(0);
      this.state.nextEpisodeRatioAnimated.setValue(0);
      this.setState({
        dyPanResponder: 0,
      });
    }
  }

  componentDidMount() {
    // MOCK PROGESS
    // this.onProgress(0);
  }

  onBookmarkPress() {
    this.props.addEpisodeBookmark(this.props.episode);
  }

  onRemoveBookmarkPress() {
    this.props.removeBookmarks([this.props.episodeBookmark.id]);
  }

  onSharePress() {
    Share.share({
      title: `EP.${this.props.episode.no} ${this.props.episode.title}`,
      message: 'http://139.59.253.62/mock-deep-link/player.html',
    });
  }

  onNextEpisode() {
    setTimeout(() => {
      this.props.refetch({
        id: this.props.episode.nextEpisode.id,
      });
    }, toMaxHeightIndicatorDuration + 100)
  }

  onProgress(ratio) {
    if (ratio >= 1 && !this.state.endedScrollView) {
      this.setState({
        endedScrollView: true,
      });
    }
    if (ratio < 1 && this.state.endedScrollView) {
      this.setState({
        endedScrollView: false,
      });
    }
  }

  onCommentPress() {
    this.props.goToComment(this.props.episode.id);
  }

  nextEpisodeRatio() {
    const diffDy = -this.state.dyPanResponder / 3;
    let ratio = diffDy / MAX_HEIGHT_INDICATOR;
    if (ratio < 0) {
      ratio = 0;
    }
    return ratio;
  }

  renderNavBarButton() {
    /* eslint class-methods-use-this: warn */
    return (
      <View style={{ flexDirection: 'row' }}>
        {
          !this.props.episodeBookmark ?
            <TouchableOpacity onPress={this.onBookmarkPress}>
              <Image style={styles.navButton} source={require('../../assets/reader/bookmark-button.png')} />
            </TouchableOpacity> :
            <TouchableOpacity onPress={this.onRemoveBookmarkPress}>
              <Image style={styles.navButton} source={require('../../assets/reader/bookmarked-button.png')} />
            </TouchableOpacity>
        }
        <TouchableOpacity>
          <Image style={styles.navButton} source={require('../../assets/reader/follow-button.png')} />
        </TouchableOpacity>
      </View>
    );
  }

  renderTitle() {
    return (<View style={styles.textContainer}>
      <Text numberOfLines={1} style={styles.titleText}>
        EP.{this.props.episode.no} {this.props.episode.title}
      </Text>
      <Text style={styles.viewCountText}>
        อ่านแล้ว {toHumanNumber(this.props.episode.viewCount)} คน
      </Text>
    </View>);
  }

  render() {
    if (this.props.loading) {
      return null;
    }

    // source={{ uri: `${serverURL}/clog/${this.props.episode.id}` }}

    return (
      <View style={{ flex: 1 }}>
        <StatusBar
          hidden={false}
        />
        <NavBarWithPinkButton
          onBackPress={this.props.onBackPress}
          renderRightMenu={this.renderNavBarButton}
          renderTitle={this.renderTitle}
          titleStyle={{
            justifyContent: 'center',
            alignItems: 'flex-start',
          }}
          leftMenuStyle={{
            flex: 0,
            paddingRight: 10,
          }}
        />
        <View style={{ flex: 1 }} {...this._panResponder.panHandlers}>
          <ModalSpinner visible={this.state.loading} />
          <View style={{flex: 1, backgroundColor: 'white'}}>
              <ClogPlayer
                source={{ uri: `${serverURL}/clog/${this.props.episode.id}` }}
                style={{ flex: 1 }}
                onMessage={() => console.log('')}
                onLoad={() => this.setState({ loading: false })}
                onProgress={this.onProgress}
            />
          </View>
          {
           this.state.nextEpisodeRatioAnimatedValue > 0 ?
              <Animated.View
                style={{
                  position: 'absolute',
                  bottom: 5,
                  opacity: this.state.nextEpisodeRatioAnimated.interpolate({
                    inputRange: [0, 0.3, 1],
                    outputRange: [0, 1, 1],
                  }),
                  left: (screenWidth / 2) - 15,
                  transform: [
                    { translateY: Animated.multiply(this.state.nextEpisodeRatioAnimated, -MAX_HEIGHT_INDICATOR) },
                  ]
                }}
              >
                <Progress.Pie progress={this.nextEpisodeRatio() > 1 ? 1 : this.nextEpisodeRatio()} size={30} />
              </Animated.View>
               : null
          }
        </View>
        <BookAndPlayerTabBar
          onSharePress={this.onSharePress}
          onCommentPress={this.onCommentPress}
          likeCount={this.props.episode.likeCount}
          commentCount={this.props.episode.commentCount}
        />
      </View>
    );
  }
}

Player.fragments = {
  episode: gql`
    fragment PlayerEpisode on Episode {
      id
      clogId
      no
      title
      likeCount
      viewCount
      nextEpisode {
        id
      }
      ...AddedBookmarkEpisode
    }
  `,
  bookmark: gql`
    fragment PlayerBookmark on Bookmark {
      id
      clogId
      episodeId
    }
  `,
};

export default Player;
