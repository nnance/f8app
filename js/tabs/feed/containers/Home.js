import gql from 'graphql-tag';
import { graphql } from 'react-apollo';

import FeedHome from '../components/FeedHome';

export const query = gql`
{
  feed: feedEditor(filter: {userId: "58b62c5fda82d62bae708b3b"}) {
    author {
      name
      profilePicture
    }
    clog {
      createdAt
      clog {
        id
        title
        coverImage
        category
        synopsis
      }
    }
  }
}
`;

const mapQueryToProps = ({ data }) => {
  const { loading, feed } = data;
  return ({
    feed: loading ? [] : feed,
    loading,
  });
};

export default graphql(query, {
  props: mapQueryToProps,
})(FeedHome);
