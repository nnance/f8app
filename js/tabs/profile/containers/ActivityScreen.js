import { connect } from 'react-redux';
import ActivityScreen from '../components/ActivityScreen';
import * as mockData from '../mockData';

export default connect(() => ({ activity: mockData.activity }))(ActivityScreen);
