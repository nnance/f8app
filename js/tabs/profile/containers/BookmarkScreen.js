import { graphql } from 'react-apollo';
import gql from 'graphql-tag';

import { updateMeBookmarksReduer, withRemoveBookmarks } from '../../../models/bookmark';

import BookmarkScreen from '../components/BookmarkScreen';

export const query = gql`
  query {
    me {
      id
      bookmarks {
        ...BookmarkScreen
      }
    }
  }
  ${BookmarkScreen.fragments.bookmark}
`;

const mapQueryToProps = ({ data }) => {
  if (data.error) {
    console.error(data.error);
  }
  return {
    bookmarks: !data.me || !data.me.bookmarks ? [] : data.me.bookmarks,
  };
};

export default withRemoveBookmarks(graphql(
  query,
  {
    props: mapQueryToProps,
    options: () => ({
      reducer: updateMeBookmarksReduer,
    }),
  },
)(BookmarkScreen));
