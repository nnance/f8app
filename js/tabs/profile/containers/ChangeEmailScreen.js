import React from 'react';
import {
  View,
  Text,
  TextInput,
  StyleSheet,
} from 'react-native';

import F8Button from 'F8Button';
import { connect } from 'react-redux';

import SecureContainer from '../components/SecureContainer';
import NavBar from '../components/NavBar';

import { styles as commonStyles } from '../common';
import { colors as commonColors } from '../../../common/styles';

import { changeEmail } from '../../../actions/changeProfile';
import ModalSpinner from '../../../common/ModalSpinner';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: commonColors.nearlyWhiteBackground,
  },
  editorContainer: {
    backgroundColor: 'white',
    marginTop: 10,
  },
  button: {
    margin: 50,
  },
});

class ChangeEmailScreen extends React.Component {
  constructor(...args) {
    super(...args);
    this.state = {
      email: '',
      confirmEmail: '',
      error: null,
      saving: false,
    };
  }

  onChangeEmail() {
    const email = this.state.email;
    const confirmEmail = this.state.confirmEmail;
    this.setState({
      error: null,
      saving: true,
    });
    if (email !== confirmEmail) {
      this.setState({
        error: 'email not match',
        saving: false,
      });
      return Promise.resolve();
    }
    return this.props.changeEmail(email).then(() => {
      this.setState({
        svaing: false,
      });
      if (this.props.onBackPress) {
        this.props.onBackPress();
      }
    }).catch((error) => {
      this.setState({
        saving: false,
        error: error.message,
      });
    });
  }

  render() {
    return (<View style={styles.container}>
      <ModalSpinner visible={this.state.saving} />
      <NavBar
        title="เปลี่ยน Email"
        onBackPress={this.props.onBackPress}
      />
      <SecureContainer onCheck={() => Promise.resolve()}>
        <View>
          <View style={styles.editorContainer}>
            <View style={commonStyles.row}>
              <View style={commonStyles.label}>
                <Text>New Email</Text>
              </View>
              <View style={commonStyles.inputContainer}>
                <TextInput
                  style={{ flex: 1, height: 40 }}
                  placeholder="email"
                  autoCapitalize="none"
                  value={this.state.email || ''}
                  onChangeText={email => this.setState({ email })}
                />
              </View>
            </View>
            <View style={commonStyles.row}>
              <View style={commonStyles.label}>
                <Text>Confirm Email</Text>
              </View>
              <View style={commonStyles.inputContainer}>
                <TextInput
                  style={{ flex: 1, height: 40 }}
                  placeholder="confirm email"
                  autoCapitalize="none"
                  value={this.state.confirmEmail || ''}
                  onChangeText={confirmEmail => this.setState({ confirmEmail })}
                />
              </View>
            </View>
          </View>
          <View style={commonStyles.errorContainer}>
            <Text style={commonStyles.errorText}>{this.state.error}</Text>
          </View>
          <F8Button style={styles.button} caption="เปลี่ยน Email" onPress={() => this.onChangeEmail()} />
        </View>
      </SecureContainer>
    </View>);
  }
}

const actionsMaping = {
  changeEmail,
};

export default connect(null, actionsMaping)(ChangeEmailScreen);
export {
  ChangeEmailScreen as Component,
};
