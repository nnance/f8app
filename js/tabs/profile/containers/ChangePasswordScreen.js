import React from 'react';
import {
  View,
  Text,
  TextInput,
  StyleSheet,
} from 'react-native';
import { connect } from 'react-redux';

import F8Button from 'F8Button';

import SecureContainer from '../components/SecureContainer';
import NavBar from '../components/NavBar';

import { styles as commonStyles } from '../common';
import { colors as commonColors } from '../../../common/styles';

import { changePassword } from '../../../actions/changeProfile';
import ModalSpinner from '../../../common/ModalSpinner';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: commonColors.nearlyWhiteBackground,
  },
  editorContainer: {
    backgroundColor: 'white',
    marginTop: 10,
  },
  button: {
    margin: 50,
  },
});

class ChangePasswordScreen extends React.Component {
  constructor(...args) {
    super(...args);
    this.state = {
      password: '',
      confirmPassword: '',
      error: null,
      saving: false,
    };
  }

  onChangePassword() {
    const password = this.state.password;
    const confirmPassword = this.state.confirmPassword;
    this.setState({
      error: null,
      saving: true,
    });
    if (password !== confirmPassword) {
      this.setState({
        error: 'password not match',
        saving: false,
      });
      return Promise.resolve();
    }
    return this.props.changePassword(password).then(() => {
      this.setState({
        saving: false,
      });
      if (this.props.onBackPress) {
        this.props.onBackPress();
      }
    }).catch((error) => {
      this.setState({
        saving: false,
        error: error.message,
      });
    });
  }

  render() {
    return (<View style={styles.container}>
      <ModalSpinner visible={this.state.saving} />
      <NavBar
        title="เปลี่ยน Password"
        onBackPress={this.props.onBackPress}
      />
      <SecureContainer onCheck={() => Promise.resolve()}>
        <View>
          <View style={styles.editorContainer}>
            <View style={commonStyles.row}>
              <View style={commonStyles.label}>
                <Text>New password</Text>
              </View>
              <View style={commonStyles.inputContainer}>
                <TextInput
                  style={{ flex: 1, height: 40 }}
                  placeholder="password"
                  autoCapitalize="none"
                  value={this.state.password || ''}
                  secureTextEntry
                  onChangeText={password => this.setState({ password })}
                />
              </View>
            </View>
            <View style={commonStyles.row}>
              <View style={commonStyles.label}>
                <Text>Confirm password</Text>
              </View>
              <View style={commonStyles.inputContainer}>
                <TextInput
                  style={{ flex: 1, height: 40 }}
                  placeholder="confirm password"
                  autoCapitalize="none"
                  value={this.state.confirmPassword || ''}
                  secureTextEntry
                  onChangeText={confirmPassword => this.setState({ confirmPassword })}
                />
              </View>
            </View>
          </View>
          <View style={commonStyles.errorContainer}>
            <Text style={commonStyles.errorText}>{this.state.error}</Text>
          </View>
          <F8Button style={styles.button} caption="เปลี่ยน Password" onPress={() => this.onChangePassword()} />
        </View>
      </SecureContainer>
    </View>);
  }
}

const actionsMaping = {
  changePassword,
};

export default connect(null, actionsMaping)(ChangePasswordScreen);
export {
  ChangePasswordScreen as Component,
};
