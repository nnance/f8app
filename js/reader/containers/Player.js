import gql from 'graphql-tag';
import { graphql } from 'react-apollo';
import { createReducerOnReply, createReducerOnDelete } from 'graphql-comment/src/react-native';

import { filterClogId, filterEpisodeId, withAddEpisodeBookmark, withRemoveBookmarks, updateMeBookmarksReduer, fragments } from '../../models/bookmark';
import Player from '../components/Player';

export const query = gql`
  query PlayerQuery($id: MongoID!){
    episode(filter: { _id: $id }) {
      ...PlayerEpisode
    }
    me {
      id
      bookmarks {
        ...PlayerBookmark
      }
    }
  }
  ${Player.fragments.episode}
  ${Player.fragments.bookmark}
  ${fragments.AddedBookmarkEpisode}
`;

export const mapQueryToProps = ({ data }) => {
  const { episode, error, loading, me } = data;
  if (error) {
    console.error('graphql error: ', error);
  }
  let episodeBookmarks;
  if (episode && me) {
    episodeBookmarks = filterEpisodeId(filterClogId(me.bookmarks, episode.clogId), episode.id);
  }

  return {
    loading,
    episode: episode || {},
    episodeBookmark: episodeBookmarks ? episodeBookmarks[0] : null,
    refetch: data.refetch,
  };
};
export const mapPropsToOptions = ({ id }) => ({
  variables: {
    id,
  },
  reducer: (previousResult, action) => {
    const deleteReducer = createReducerOnDelete(id, (previousResult) => {
      return {
        ...previousResult,
        episode: {
          ...previousResult.episode,
          commentCount: previousResult.episode.commentCount - 1,
        },
      };
    });
    const replyReducer = createReducerOnReply(id, (previousResult, reply) => {
      return {
        ...previousResult,
        episode: {
          ...previousResult.episode,
          commentCount: previousResult.episode.commentCount + 1,
        },
      };
    });
    return deleteReducer(replyReducer(updateMeBookmarksReduer(previousResult, action), action), action);
  },
});

const withData = graphql(query, {
  props: mapQueryToProps,
  options: mapPropsToOptions,
});

export default withData(withRemoveBookmarks(withAddEpisodeBookmark(Player)));
