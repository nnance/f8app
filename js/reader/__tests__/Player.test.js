/* eslint import/first: off */
jest.mock('react-native-fs', () => null);

import React from 'react';
import renderer from 'react-test-renderer';

import graphql from '../../libs/mockGraphQL';
import { query, mapQueryToProps, mapPropsToOptions } from '../containers/Player';
import Player from '../components/Player';

describe('Player', () => {
  it('render loading', () => {
    const tree = renderer.create(<Player loading />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('render', async () => {
    const id = '58b95c905923032426ceb6ce';
    const result = await graphql(query, mapPropsToOptions({ id }));
    const props = mapQueryToProps({ data: result.data });
    const tree = renderer.create(<Player id={id} {...props} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
