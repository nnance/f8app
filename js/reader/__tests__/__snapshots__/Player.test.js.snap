exports[`Player render 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-between",
          "paddingTop": 15,
          "width": 750,
        },
        undefined,
      ]
    }>
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "paddingLeft": 10,
          },
          Object {
            "flex": 0,
            "paddingRight": 10,
          },
        ]
      }>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}>
        <Image
          source={1}
          style={
            Object {
              "height": 20,
              "resizeMode": "contain",
            }
          } />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 4,
          },
          Object {
            "alignItems": "flex-start",
            "justifyContent": "center",
          },
        ]
      }>
      <View
        style={
          Object {
            "marginRight": 50,
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "rgb(249, 65, 92)",
              "fontSize": 14,
            }
          }>
          EP.
          99
           
          Asperiores quisquam
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(249, 65, 92, 0.6)",
              "fontSize": 10,
            }
          }>
          อ่านแล้ว 
          0
           คน
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "flex": 1,
            "paddingRight": 10,
          },
          undefined,
        ]
      }>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}>
          <Image
            source={1}
            style={
              Object {
                "height": 25,
                "marginRight": 10,
                "resizeMode": "contain",
                "width": 25,
              }
            } />
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}>
          <Image
            source={1}
            style={
              Object {
                "height": 25,
                "marginRight": 10,
                "resizeMode": "contain",
                "width": 25,
              }
            } />
        </View>
      </View>
    </View>
  </View>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }>
    <Modal
      transparent={true}
      visible={true}>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(255, 255, 255, 0.6)",
            "flex": 1,
            "justifyContent": "center",
          }
        }>
        <ActivityIndicator
          animating={true}
          color="#999999"
          hidesWhenStopped={true}
          size="large"
          style={
            Array [
              Object {
                "height": 80,
              },
            ]
          } />
      </View>
    </Modal>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
      Loading
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 45,
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "resizeMode": "contain",
          }
        } />
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1.6,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "resizeMode": "contain",
          }
        } />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.3)",
          }
        }>
        100
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1.6,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "resizeMode": "contain",
          }
        } />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.3)",
          }
        }>
        0
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "resizeMode": "contain",
          }
        } />
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "resizeMode": "contain",
          }
        } />
    </View>
  </View>
</View>
`;

exports[`Player render loading 1`] = `null`;
